pkg/loop/median_service.go

```diff
@@ -0,0 +1,42 @@
+package loop
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2/types"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/logger"
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+var _ PluginMedian = (*MedianService)(nil)
+
+// MedianService is a [types.Service] that maintains an internal [PluginMedian].
+type MedianService struct {
+	*pluginService[*GRPCPluginMedian, PluginMedian]
```

Comment:
does `Plugin` mean the same thing in these two names? are they both loops or one a loop and the other an ocr?

Reply:
I'm not sure that I understand the question. The median loop plugin uses the same interfaces defined by libocr. The two types passed here are for the go-plugin type and our own custom type.

Reply:
we use plugin to mean libocr plugin and loop plugin.

it's super hard to keep these different meaning straight. what would it looks to stop using the word Plugin in new dev? could use something like Loop/Loopp/Loopper and OCRReporter

Reply:
I don't think that playing with those terms simplifies the problem in this case. We are already stuck with hashicorp/go-plugin and libocr ReportingPlugin which are inseparable. packages should help, but we're always going to have code that wraps and implements on both sides.